### DISCUSSION TITLE
Web Key Directory (WKD)

### DISCUSSION THREAD

**Discussion by almereyda** [modoboa/modoboa#1443] [Ideas]
# Impacted versions

* Modoboa: all

# Steps to reproduce

Users of Enigmail 2.0 are checking a *Web Key Directory* (WKD) service for available public keys of user mailboxes.

# Current behavior

They will not receive the keys from Modoboa servers.

# Expected behavior

Modoboa provides a WKD for the mailaccounts and domains it handles.

# link

See https://wiki.gnupg.org/WKD for the specification.

**Comment by wiktor-k** [comment]
How would the key <-> username association be created?

Several options:
  1. [Web Key Service](https://datatracker.ietf.org/doc/draft-koch-openpgp-webkey-service/), basically a challenge-response mechanism through e-mail,
  2. Putting user's fingerprint in configuration and automatically fetching keys from keyservers, putting them in a well-known location,
  3. Putting user's keys directly in config.
